lib/elixir/lib/integer.ex

```diff
@@ -240,45 +240,30 @@ defmodule Integer do
   end
 
   def parse(<<?-, rest::binary>>, base) do
-    case parse_digits(rest, base) do
-      {acc, bin} ->
-        {-acc, bin}
-      :error ->
-        :error
-    end
-  end
-
-  def parse(<<?+, rest::binary>>, base) do
-    parse_digits(rest, base)
+    with {int, rem} <- parse_digits(rest, base), do: {-int, rem}
   end
-
-  def parse(<<rest::binary>>, base) do
-    parse_digits(rest, base)
-  end
-
-  digits = [{?0..?9, -?0}, {?A..?Z, 10 - ?A}, {?a..?z, 10 - ?a}]
-
-  for {chars, diff} <- digits, char <- chars do
-    defp parse_digits(<<unquote(char), rest::binary>>, base)
-         when base > unquote(char + diff) do
-      parse_digits(rest, base, unquote(char + diff))
+  def parse(<<?+, rest::binary>>, base), do: parse_digits(rest, base)
+  def parse(<<rest::binary>>, base), do: parse_digits(rest, base)
+
+  defp parse_digits(<<rest::binary>>, base) do
+    case count_digits(rest, base, 0) do
+      0 -> :error
+      n ->
+        {digits, rem} = :erlang.split_binary(rest, n)
```

Comment:
I'm confused. First, the `parse_digits` function that you're referring to doesn't do anything related handling +/-. That's handled higher in the call stack, in the `def parse(<<?-, rest::binary>>, base)` and `def parse(<<?+, rest::binary>>, base)` function definitions. Also, moving the sign match into the recursive `count_digits` would cause bugs if an input like `"1-2+3"` were passed.

Can you please provide a code example of what you mean?